packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -19,7 +19,8 @@ interface FocusScopeProps {
   children: ReactNode,
   contain?: boolean,
   restoreFocus?: boolean,
-  autoFocus?: boolean
+  autoFocus?: boolean,
+  setActiveScope?: boolean
```

Comment:
API change?

Reply:
also, `isActiveScope`?

Reply:
`replaceScope`?

Reply:
suggesting other names because setActiveScope collides with hook usage in my head, sounds like setState conventions

Reply:
yep, added this api in since the issue is that the nested Dialog's FocusScope isn't set as the active scope on mount. Then when it attempts to focus the nested dialog, the first FocusScope detects that the element is not in its scope and resets focus back to the button that triggered the nested dialog.

I like `isActiveScope`, I'll go and use that

Reply:
can you bisect and determine when this broke? it did used to work.

Reply:
Yup, it used to work cuz we had autofocus on the Dialogs by default but it was removed in a prior commit (952e102a3e162f64a1009b0e4305dbd563ffd053) since we didn't want the Dialog to autoscroll to a link if it was deep down in a long body of text in the Dialog.

Reply:
hmm... autofocus shouldn't affect blurring outside the scope though right?

Reply:
I believe the reason autofocus made this work is because it updated the `activeScope` to be equal to the nested dialog's `scopeRef` so that when useDialog tries to focus the nested dialog on mount, the focus isn't perceived as a event outside of the active scope, thus it doesn't reset the focus to the previously focused element (i.e. the dialog trigger button)

Reply:
ah, that seems like the real issue. we do autofocus the dialog itself though. we should fix it so the active scope is correctly set when that happens.

Reply:
can we find out why the active scope isn't set properly? we do auto focus the dialog itself, which is inside the scope, so it should handle that.

Reply:
So the `activeScope` is only updated in two cases, the first being if `autoFocus` is set on one of the `FocusScope` (inapplicable in this case) and the second being from `onFocus` when a focused element is found in the current scope and that scope is not the active scope. The problematic flow is as follows:
1. The first dialog is opened and focus is called on it via `useDialog` useEffect. 
2. Its `FocusScope`'s `onFocus` is called. Since the dialog is within scope and there isn't an `activeScope` set, the first dialog's `FocusScope` is set as the `activeScope`.
3. User clicks on the action button to open the nested dialog. `onFocus` for the first `FocusScope` is called and it saves the action button as the `focusedNode`. `activeScope` doesn't change.
4. The nested dialog renders and focus is called on it via `useDialog` useEffect. `onFocus` for the **first** `FocusScope` triggers. It determines that the focus target (the nested dialog) is **not** in it's scope and since it is the current `activeScope` it resets focus back to the last `focusedNode` aka the action button

If the `onFocus` for the nested Dialog's `FocusScope` triggered first then the `activeScope` would get updated appropriately to be the nested Dialog's `FocusScope`. Is there a way to make it trigger first? Dunno if that was possible hence why I opted for the current solution

Reply:
Ah I see. I think we need to make child focus scopes take precedence over parent focus scopes. If focus occurs in a child focus scope, that scope should always become the active scope rather than the parent taking back focus.

Reply:
Mmmm, ok. I'll see if I can check the focus context and set the focus scope based on whether or not there is anything returned (indicating if it is a child focus scope)